src/Symfony/Component/HttpClient/Response/StreamWrapper.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Allows turning ResponseInterface instances to PHP streams.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class StreamWrapper
+{
+    /** @var resource */
+    public $context;
+
+    /** @var HttpClientInterface */
+    private $client;
+
+    /** @var ResponseInterface */
+    private $response;
+
+    /** @var resource|null */
+    private $content;
+
+    /** @var resource|null */
+    private $handle;
+
+    private $eof = false;
+    private $offset = 0;
+
+    /**
+     * Creates a PHP stream resource from a ResponseInterface.
+     *
+     * @param resource|null $contentBuffer The seekable resource where the response body is buffered
+     * @param resource|null $selectHandle  The resource handle that should be monitored when
+     *                                     stream_select() is used on the created stream
+     *
+     * @return resource
+     */
+    public static function createResource(ResponseInterface $response, HttpClientInterface $client = null, $contentBuffer = null, $selectHandle = null)
+    {
+        if (null === $client && !method_exists($response, 'stream')) {
+            throw new \InvalidArgumentException(sprintf('Providing a client to "%s()" is required when the response doesn\'t have any "stream()" method.', __CLASS__));
+        }
+
+        if (false === stream_wrapper_register('symfony', __CLASS__, STREAM_IS_URL)) {
+            throw new \RuntimeException(error_get_last()['message'] ?? 'Registering the "symfony" stream wrapper failed.');
+        }
+
+        try {
+            $context = [
+                'client' => $client ?? $response,
+                'response' => $response,
+                'content' => $contentBuffer,
+                'handle' => $selectHandle,
+            ];
+
+            return fopen('symfony://'.$response->getInfo('url'), 'r', false, stream_context_create(['symfony' => $context])) ?: null;
+        } finally {
+            stream_wrapper_unregister('symfony');
+        }
+    }
+
+    public function stream_open(string $path, string $mode, int $options): bool
```

Comment:
Is this purposeful that the methods are `stream_open` and not `streamOpen` . I see this is mixed now . Accidental ?

Reply:
My bad, I see where this is ending up. https://www.php.net/manual/en/stream.streamwrapper.example-1.php . Sorry about the comment.